syntax = "proto3";
package aumos.api.v1;

import "google/protobuf/timestamp.proto";

message PaginationRequest {
  int32 page = 1;
  int32 page_size = 2;
  string sort_by = 3;
  string sort_order = 4;
}

message PaginationResponse {
  int32 total = 1;
  int32 page = 2;
  int32 page_size = 3;
  int32 pages = 4;
  bool has_next = 5;
  bool has_prev = 6;
}

message ErrorResponse {
  string code = 1;
  string message = 2;
  map<string, string> details = 3;
  string request_id = 4;
  google.protobuf.Timestamp timestamp = 5;
}
